File: src/main/java/ru/mpu/horse_races/services/OwnerService.java
package ru.mpu.horse_races.services;

import ru.mpu.horse_races.domain.dtos.CreateOrUpdateOwnerDtoRq;
import ru.mpu.horse_races.domain.dtos.OwnerDto;

import java.util.List;

public interface OwnerService {
    OwnerDto insert(CreateOrUpdateOwnerDtoRq owner);

    OwnerDto update(CreateOrUpdateOwnerDtoRq owner);

    List<OwnerDto> findAll();
}

File: frontend/horse-races/src/app/owner/owner.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CreateOrUpdateOwnerDtoRq, OwnerDto } from './owner.models';

@Injectable({
  providedIn: 'root'
})
export class OwnerService {
  private apiUrl = '/api/v1/owner';

  constructor(private http: HttpClient) {}

  findAll(): Observable<OwnerDto[]> {
    return this.http.get<OwnerDto[]>(this.apiUrl);
  }

  insert(owner: CreateOrUpdateOwnerDtoRq): Observable<OwnerDto> {
    return this.http.post<OwnerDto>(this.apiUrl, owner);
  }

  update(owner: CreateOrUpdateOwnerDtoRq): Observable<OwnerDto> {
    return this.http.put<OwnerDto>(`${this.apiUrl}/${owner.id}`, owner);
  }

  deleteById(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
